src/gpu/intel/sycl/interop_kernel.cpp

```diff
@@ -144,6 +144,7 @@ status_t interop_kernel_t::parallel_for(impl::stream_t &stream,
                             auto *m = utils::downcast<
                                     const xpu::sycl::usm_memory_storage_t *>(
                                     mem_storage);
+                            CHECK(check_alignment(m->usm_ptr()));
```

Comment:
Wouldn't that prevent reference implementation from executing as well?

Reply:
Good catch, thanks. I updated to check the kernel name as well. Let's assume all kernels named with "ref_*" do not use block access and do not have any specific alignment requirements.